### PR TITLE
feat(koa-cash): add method from v4.1.1

### PR DESCRIPTION
**Pull request by rhuanbarreto**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [<<url here>>](https://github.com/koajs/cash/pull/98)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by jagoe** [comment]
@rhuanbarreto Sorry for the late response. It has been a while since I added the typings, so I'm not a 100% sure, but your contribution looks good to me.

According to the `test` action, you're not allowed to specify a patch version in the type definition header. Pretty sure that's because patch versions aren't supposed to change public interfaces (https://semver.org).
So just remove the patch version (`4.1` instead of `4.1.1`) and you should be good to go.

File: types/koa-cash/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for koa-cash 4.0
+// Type definitions for koa-cash 4.1.1
```
Comment (rhuanbarreto) [inline types/koa-cash/index.d.ts:1]:
```suggestion
// Type definitions for koa-cash 4.1
```

**Comment by rhuanbarreto** [comment]
@jagoe Changes made. You can approve and merge.

File: types/koa-cash/koa-cash-tests.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ app.use(koaCash());
 app.use(async ctx => {
     // this response is already cashed if `true` is returned,
     // so this middleware will automatically serve this response from cache
-    if (await ctx.cashed()) return;
+    if (await ctx.cashed()) return ctx.cashClear('/');
```
Comment (jagoe) [inline types/koa-cash/koa-cash-tests.ts:10]: Wasn't sure at first, but the latest build error confirms it. `cashClear()` returns `void`, so this doesn't make sense. Maybe something like this:

```suggestion
    if (await ctx.cashed()) {
      ctx.cashClear('/');
      return;
    }
```

Wouldn't make much sense for an application, but whatever :smile:

**Review by jagoe — CHANGES_REQUESTED** [review]

**Review by jagoe — APPROVED** [review]

**Comment by rhuanbarreto** [comment]
Ready to merge

**Pull request merged**